docs/usage.rst

```diff
@@ -2,4 +2,213 @@
 Usage
 =====
 
-…
+:ref:`Given an HTML form <form>`:
+
+.. _fromstring-example:
+
+>>> from lxml.html import fromstring
+>>> html = b"""<form><input type="hidden" name="foo" value="bar" /></form>"""
+>>> root = fromstring(html, base_url="https://example.com")
+>>> form = root.xpath("//form")[0]
+
+You can use :func:`~form2request.request_from_form` to generate :ref:`form
+submission request data <request>`:
+
+>>> from form2request import request_from_form
+>>> request_from_form(form)
+Request(url='https://example.com?foo=bar', method='GET', headers=[], body=b'')
+
+:func:`~form2request.request_from_form` supports :ref:`user-defined form data
+<data>` and :ref:`choosing a specific form submission button (or none)
+<click>`.
+
+
+.. _form:
+
+Getting a form
+==============
+
+:func:`~form2request.request_from_form` requires an
+:class:`lxml.html.FormElement` object.
+
+You can build one using :func:`lxml.html.fromstring` to parse an HTML document
+and :meth:`lxml.html.HtmlElement.xpath` to find a form element in that
+document, as :ref:`seen above <fromstring-example>`.
+
+Here are some examples of XPath expressions that can be useful to find a form
+element using :meth:`~lxml.html.HtmlElement.xpath`:
+
+-   To find a form by one of its attributes, such as ``id`` or ``name``, use
+    ``//form[@<attribute>="<value>"]``. For example, to find ``<form id="foo"
+    …``, use ``//form[@id="foo"]``.
+
+-   To find a form by index, by order of appearance in the HTML code, use
+    ``(//form)[n]``, where ``n`` is a 1-based index. For example, to find the
+    2nd form, use ``(//form)[2]``.
+
+If you prefer, you could use the XPath of an element inside the form, and then
+visit parent elements until you reach the form element. For example:
+
+.. code-block:: python
+
+    element = root.xpath('//input[@name="zip_code"]')[0]
+    while True:
+        if element.tag == "form":
+            break
+        element = element.getparent()
+    form = element
+
+If you use an lxml-based library or framework, chances are they also let you
+get a :class:`~lxml.html.FormElement` object. For example, when using
+:doc:`parsel <parsel:index>`:
+
+>>> from parsel import Selector
+>>> selector = Selector(body=html, base_url="https://example.com")
+>>> form = selector.css("form")[0].root
+>>> type(form)
+<class 'lxml.html.FormElement'>
+
+A similar example, with a :doc:`Scrapy <scrapy:index>` response:
+
+>>> from scrapy.http import TextResponse
+>>> response = TextResponse("https://example.com", body=html)
+>>> form = response.css("form")[0].root
+>>> type(form)
+<class 'lxml.html.FormElement'>
+
+
+.. _data:
+
+Setting form data
```

Comment:
Love the docs and the explanations, great work! It's something which is currently missing in Scrapy's FormRequest docs.